### PR TITLE
Rename concat_column to concatenated_categorical_column.

### PR DESCRIPTION
**Pull request by brightcoder01**
Add more description and unit tests.

### CODE REVIEW

File: elasticdl/python/elasticdl/feature_column/feature_column.py [hunk 1]
```diff
@@ -210,7 +210,80 @@ def embedding_weight_name(self):
...
+
+    Use this when you have many categorical columns and want to map all
+    to embedding. It's recommended to concatenate the sparse id tensors
+    from the categorical columns above to one sparse id tensor using this
+    API, and then map the combined SparseTensor to Embedding. Because
+    there is much overhead to create embedding variables for each
+    categorical column, the model size will be huge. Using this way, we
+    can reduce the model size.
+
+    The id range of source categorical columns are [0, num_buckets_0],
```
Comment (workingloong) [inline elasticdl/python/elasticdl/feature_column/feature_column.py]: "The id range of " -> "The output integer id range of"

Comment (workingloong) [reply elasticdl/python/elasticdl/feature_column/feature_column.py]: Why is it `[0, num_buckets_0]` not `[0, num_buckets_0)`

Comment (brightcoder01) [reply elasticdl/python/elasticdl/feature_column/feature_column.py]: > Why is it `[0, num_buckets_0]` not `[0, num_buckets_0)`

Good catch, will update it.

**Review by terrytangyuan — APPROVED** [review]
LGTM once comments are addressed

File: elasticdl/python/elasticdl/feature_column/feature_column.py (hunk 1 above)
Comment (brightcoder01) [reply elasticdl/python/elasticdl/feature_column/feature_column.py]: > "The id range of " -> "The output integer id range of"

Updated to `The output id range of`

**Pull request merged**